File: week1/problemset1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int bricks);
void print_bricks(int bricks, string brick);

int main(void)
{
    int height;

    do
    {
        height = get_int("Height: ");
    }
    while(height < 1 || height > 8);

    for(int i = 1; i <= height; i++)
    {
        print_row(height - i, i);
    }
}

void print_row(int spaces, int bricks)
{
    print_bricks(spaces, " ");

    print_bricks(bricks, "#");

    printf("  ");

    print_bricks(bricks, "#");

    printf("\n");
}

void print_bricks(int bricks, string brick){
    for (int i = 0; i < bricks; i++)
    {
        printf("%s", brick);
    }
}

File: week6/problemset/sentimental-mario-less/mario.py
def main():
    height = get_height()

    print_pyramide(height)


def print_pyramide(height):
    for i in range(height):
        print_row(height - i - 1, i + 1)


def print_row(spaces, blocks):
    print(" " * spaces + "#" * blocks)


def get_height():
    while True:
        height = get_int("Height ")

        if height > 0 and height <= 8:
            break
        else:
            print("Error: Must be a positive Integer lesser or equal than 8")

    return height


def get_int(prompt):
    while True:
        try:
            return int(input(prompt))
        except ValueError:
            print("Not a Integer")


main()
